### PR TITLE
Implemented functionality to fetch and show order details in orderDetails page ( #1x634cr )

### PR DESCRIPTION
**Pull request by meet-aniket**
Screenshot from 2021-12-27 18-13-41
Screenshot from 2021-12-27 18-13-16

### CODE REVIEW

File: src/store/index.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ const state: any = {
 }
 
 const persistState = createPersistedState({
-    paths: ['user'],
+    paths: ['user', 'product'],
```
Comment (adityasharma7) [inline src/store/index.ts]: I think we should not add product to storage

File: src/store/modules/order/actions.ts [hunk 2]
```diff
@@ -36,6 +37,67 @@ const actions: ActionTree<OrderState, RootState> = {
...
+        "query": "*:*",
+        "filter": `docType: ORDER AND orderTypeId: SALES_ORDER AND orderId: ${orderId}`
+      }
+    }
+
+    try {
+      resp = await OrderService.findOrderDetails(payload);
+
+      if (resp.status == 200 && !hasError(resp)) {
+        const orderDetails: Order = {
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts]:
```suggestion
        const order: Order = {
```

File: src/store/modules/order/mutations.ts [hunk 3]
```diff
@@ -6,6 +6,9 @@ const mutations: MutationTree <OrderState> = {
   [types.ORDER_LIST_UPDATED] (state, payload) {
     state.list.orders = payload.orders
     state.list.total = payload.total
+  },
+  [types.ORDER_CURRENT_UPDATED] (state, payload) {
+    state.current = payload.orderDetails
```
Comment (adityasharma7) [inline src/store/modules/order/mutations.ts]:
```suggestion
    state.current = payload.order
```

File: src/views/Order.vue [hunk 4]
```diff
@@ -44,34 +47,35 @@
...
 
           <div class="info">
             <ion-card>
               <ion-list>
                 <ion-item lines="none">
-                  <ion-label>John Doe</ion-label>
-                  <ion-chip slot="end">
+                  <ion-label> {{ order.customer?.name }} </ion-label>
+                  <!-- TODO: handle this property to display loyalty options -->
+                  <ion-chip slot="end" v-if="order.cusotmerLoyaltyOptions">
```
Comment (adityasharma7) [inline src/views/Order.vue]: I think this should be handled with a filter, check preorder app for reference

File: src/views/Order.vue [hunk 5]
```diff
@@ -81,15 +85,15 @@
                 <ion-list-header>{{ $t("Shopify IDs") }}</ion-list-header>
                 <ion-item>
                   <ion-label> {{ $t("Order Number") }} </ion-label>
-                  <p slot="end">14 Jan 2021</p>
+                  <p slot="end">{{ $filters.getIdentification(order.identifications, orderNo) ? $filters.getIdentification(order.identifications, orderNo) : "-" }}</p>
```
Comment (adityasharma7) [inline src/views/Order.vue]: We could process it while setting

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/index.ts (hunk 1 above)
Comment (ymaheshwari1) [reply src/store/index.ts]: Sir, removed the store from persisting

File: src/store/modules/order/actions.ts (hunk 2 above)
Comment (ymaheshwari1) [reply src/store/modules/order/actions.ts]: Improved the naming convention from orderDetails to order.

File: src/store/modules/order/mutations.ts (hunk 3 above)
Comment (ymaheshwari1) [reply src/store/modules/order/mutations.ts]: Done Sir

File: src/views/Order.vue (hunk 4 above)
Comment (ymaheshwari1) [reply src/views/Order.vue]: Added a filter to correctly handle the customerLoyaltyOptions.

File: src/views/Order.vue (hunk 5 above)
Comment (ymaheshwari1) [reply src/views/Order.vue]: Processed the order identification values when getting the response.

File: src/views/Order.vue [hunk 6]
```diff
@@ -154,62 +155,71 @@
...
+                      </ion-buttons> -->
                     </ion-list>
                   </ion-card>
                   <ion-card>
                     <ion-list>
                       <ion-list-header>{{ $t("Fulfillment") }}</ion-list-header>
                       <ion-item>
                         <ion-label> {{ $t("Shipping method") }} </ion-label>
-                        <p>Standard</p>
+                        <p>{{ item.shipmentMethodTypeId ? item.shipmentMethodTypeId : "-"}}</p>
```
Comment (adityasharma7) [inline src/views/Order.vue]: This should be the shipment method description
We need to gather this info from ShipmentMethodType entity

File: src/views/Order.vue [hunk 7]
```diff
@@ -44,34 +47,35 @@
...
 
           <div class="info">
             <ion-card>
               <ion-list>
                 <ion-item lines="none">
-                  <ion-label>John Doe</ion-label>
-                  <ion-chip slot="end">
+                  <ion-label> {{ order.customer?.name }} </ion-label>
+                  <!-- TODO: handle this property to display loyalty options -->
+                  <ion-chip slot="end" v-if="$filters.getCustomerLoyalty(order.orderNotes, cusotmerLoyaltyOptions)">
```
Comment (adityasharma7) [inline src/views/Order.vue]: I think we can have the same approach as identifications for customer loyalty as well.
WDYT?

File: src/store/modules/order/actions.ts [hunk 8]
```diff
@@ -36,6 +37,77 @@ const actions: ActionTree<OrderState, RootState> = {
...
+          statusId: '',
+          identifications: {},
+          notes: []
+        }
+
+        const orderName = process.env.VUE_APP_ORD_IDENT_TYPE_NAME
+        const orderId = process.env.VUE_APP_ORD_IDENT_TYPE_ID
+        const orderNo = process.env.VUE_APP_ORD_IDENT_TYPE_NO
+
+        resp.data.grouped.orderId.groups.map((group: any) => {
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts]: Instead of iterating with map(), can we just get the first item in groups and populate values

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/Order.vue (hunk 7 above)
Comment (ymaheshwari1) [reply src/views/Order.vue]: Ok Sir, will create a helper for the same.

Comment (ymaheshwari1) [reply src/views/Order.vue]: Done Sir

File: src/store/modules/order/actions.ts (hunk 8 above)
Comment (ymaheshwari1) [reply src/store/modules/order/actions.ts]: Sir, then can we use something like
```
const items = resp.data.grouped.orderId.groups[0].doclist.docs
order.orderId = items[0].orderId
...
```

Comment (adityasharma7) [reply src/store/modules/order/actions.ts]: Yes, check the length of resp.data.grouped.orderId.groups and store first item in a variable. Use it further to extract data in order object

Comment (ymaheshwari1) [reply src/store/modules/order/actions.ts]: Sir updated the code.

File: src/views/Order.vue (hunk 6 above)
Comment (ymaheshwari1) [reply src/views/Order.vue]: Sir, created a state for storing the shipment methods and use that on details page

File: src/views/Order.vue [hunk 9]
```diff
@@ -290,28 +293,53 @@ export default {
...
     }
   },
-  methods: {
-    changeStatus (ev) {
+  computed: {
+    ...mapGetters({
+      order: 'order/getCurrentOrder',
+      getProduct: 'product/getProduct',
+      getProductStock: 'stock/getProductStock',
+      getShipmentMethodDescription: 'util/getShipmentMethodDesc'
```
Comment (adityasharma7) [inline src/views/Order.vue]: Either the getter should return description directly or name should improved

Comment (adityasharma7) [inline src/views/Order.vue]: Either the getter should return description directly or name should improved

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/Order.vue (hunk 9 above)
Comment (ymaheshwari1) [reply src/views/Order.vue]: Changed method name to `getShipmentMethod` and return the shipment desc from the getter only.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**